### PR TITLE
entities

### PR DESCRIPTION
**Pull request by udit-pandey**
fixed issue:(entities that are part of stories are lost when uploading training files with stories having entities)

### CODE REVIEW

File: kairon/data_processor/processor.py [hunk 1]
```diff
@@ -853,7 +853,12 @@ def __prepare_training_slots(self, bot: Text):
     def __extract_story_events(self, events):
         for event in events:
             if isinstance(event, UserUttered):
-                yield StoryEvents(type=event.type_name, name=event.intent_name)
+                entities = [Entity(
```
Comment (sfahad1414) [inline kairon/data_processor/processor.py:856]: you need to loop through entities recieved in UserUttered

File: kairon/data_processor/processor.py [hunk 2]
```diff
@@ -904,6 +909,11 @@ def __save_stories(self, story_steps, bot: Text, user: Text):
     def __prepare_training_story_events(self, events, timestamp):
         for event in events:
             if event.type == UserUttered.type_name:
+                entities = [{"start": entity['start'],
```
Comment (sfahad1414) [inline kairon/data_processor/processor.py:912]: you the pass the entire entites list recieved from the Mongo

File: kairon/data_processor/processor.py [hunk 3]
```diff
@@ -1053,8 +1064,8 @@ def add_training_data(self, training_data: List[models.TrainingData], bot: Text,
             story_name = "path_" + data.intent
             utterance = "utter_" + data.intent
             events = [
-                {"name": data.intent, "type": "user"},
-                {"name": utterance, "type": "action"}]
+                {"name": data.intent, "type": "user", "entities":[]},
+                {"name": utterance, "type": "action", "entities":[]}]
```
Comment (sfahad1414) [inline kairon/data_processor/processor.py:1068]: user action does not have entities field so entities in the db must be null

**Review by sfahad1414 — CHANGES_REQUESTED** [review]
please do the changes

File: kairon/data_processor/processor.py (hunk 3 above)
Comment (udit-pandey) [reply kairon/data_processor/processor.py:1068]: no it has empty list

File: kairon/data_processor/processor.py (hunk 2 above)
Comment (udit-pandey) [reply kairon/data_processor/processor.py:912]: cannot be passed as it is mongo object while training data requires a dict

**Comment by sfahad1414** [comment]
merging the changes

**Pull request merged**